src/entity/static_mesh.cpp

```diff
@@ -59,12 +66,52 @@ StaticMesh::StaticMesh(
         GL_ELEMENT_ARRAY_BUFFER, indices.size() * sizeof(unsigned short),
         indices.data(), GL_STATIC_DRAW
     );
+    assert(glGetError() == GL_NO_ERROR);
+
+    // Generate a texture
+    std::vector<std::array<float, 4>> float_colors;
+    for( uint32_t int_color: color_map){
+        uint32_t red = (int_color >> 24) & 0xFF;
+        uint32_t green = (int_color >> 16) & 0xFF;
+        uint32_t blue = (int_color >> 8) & 0xFF;
+        uint32_t alpha = (int_color) & 0xFF;
+        // the last one >> 0 is A
+        std::array<float, 4> vector_color({
+            red / 255.0f, green / 255.0f, blue / 255.0f, alpha / 255.0f
+        });
+        float_colors.push_back(vector_color);
+    }
+
+    assert(glGetError() == GL_NO_ERROR);
+    LOG_DEBUG(logging::opengl_logger, "float_colors {}", float_colors);
+    //LOG_DEBUG(logging::opengl_logger, "color texture id before {}", color_texture_);
+    glGenTextures(1, &color_texture_);
+    assert(glGetError() == GL_NO_ERROR);
+    //LOG_DEBUG(logging::opengl_logger, "color texture id after binding {}", color_texture_);
+    glBindTexture(GL_TEXTURE_1D, color_texture_);
+    assert(glGetError() == GL_NO_ERROR);
+    glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
+    assert(glGetError() == GL_NO_ERROR);
+    // set the texture wrapping/filtering options (on the currently bound texture object)
+    glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+    assert(glGetError() == GL_NO_ERROR);
+    glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+    assert(glGetError() == GL_NO_ERROR);
+    glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR);
+    assert(glGetError() == GL_NO_ERROR);
+    glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+    assert(glGetError() == GL_NO_ERROR);
+    // load and generate the texture
+    glTexImage1D(GL_TEXTURE_1D, 0, GL_RGBA32F, float_colors.size(), 0, GL_RGBA, GL_FLOAT, float_colors.data());
+    assert(glGetError() == GL_NO_ERROR);
+    glGenerateMipmap(GL_TEXTURE_1D);
+    //assert(glGetError() == GL_NO_ERROR);
```

Comment:
Air out this code (add newlines after each assert and before comments) and delete commented code.

Reply:
gl error handling needs to change significantly. I couldn't get it to work and wanted you to look at it.